### PR TITLE
Add StyleLintBear

### PR DESCRIPTION
**Pull request by dracarys09**
StyleLintBear is a bear which uses stylelint to lint css files.

Closes https://github.com/coala/coala-bears/issues/1255

### CODE REVIEW

**Comment by dracarys09** [comment]
I'm getting 100% coverage locally... however circle ci is showing that line 31 has some problem... can someone suggest something... ?

File: bears/css/StyleLintBear.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Detect errors and potential problems in CSS code and to enforce
+    appropriate coding conventions. For example, problems like syntax errors,
+    invalid color codes etc can be detected.
+
+    For more information on the analysis visit <http://stylelint.io/>
+    """
+    LANGUAGES = {'CSS'}
+    REQUIREMENTS = {PipRequirement('stylelint', '7.7')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
```
Comment (vijeth-aradhya) [inline bears/css/StyleLintBear.py]: Also, include ASCIINEMA URL which shows a test working of your bear.

File: bears/css/StyleLintBear.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    LICENSE = 'AGPL-3.0'
```
Comment (vijeth-aradhya) [inline bears/css/StyleLintBear.py]: Um it's good if you include `CAN_DETECT` as well, other than the place where you have mentioned in the comments.

**Comment by dracarys09** [comment]
@vijeth-aradhya I added stylelint in bear-requirements.txt, however now it is saying
 No matching distribution found for stylelint (from -r bear-requirements.txt (line 34))
Locally I used ```npm install stylelint``` to install stylelint (the linter which I'm using for this LinterBear)

Am I doing something wrong ?

File: bears/css/StyleLintBear.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    CAN_DETECT = {'Syntax', 'Unused Code', 'Formatting'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         stylelint_config_file: str=os.devnull):
+
+        return '--config=' + stylelint_config_file, filename
```
Comment (rhemon) [inline bears/css/StyleLintBear.py]: your coverage test is failing because there is no test for this

File: bears/css/StyleLintBear.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import os
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.NpmRequirement import NpmRequirement
+
+
+@linter(executable='stylelint',
+        output_format='regex',
+        output_regex=r'.+\n\s(?P<line>\d+):(?P<column>\d+)'
+                     r'\s+✖\s(?P<message>.+)')
```
Comment (aptrishu) [inline bears/css/StyleLintBear.py]: @dracarys09  IMO this regex will not cover every message, they also have a warning symbol in place of cross. Can you check ?
and tests are not sufficient.

File: bears/css/StyleLintBear.py (hunk 3 above)
Comment (dracarys09) [inline bears/css/StyleLintBear.py]: so adding a test case say ```test_create_arguments``` is a possible solution for this ?

File: bears/css/StyleLintBear.py (hunk 4 above)
Comment (dracarys09) [inline bears/css/StyleLintBear.py]: @aptrishu I'll add more test cases... however I can't figure out what changes I need to make in `bear-requirements.txt` and `.ci/deps.sh`... can you help me out with these ?

Comment (vijeth-aradhya) [inline bears/css/StyleLintBear.py]: @dracarys09 Why don't you install it the same way in CI you did locally, https://github.com/coala/coala-bears/blob/master/.ci/deps.sh#L101? Did you try that?

File: bears/css/StyleLintBear.py (hunk 3 above)
Comment (Asnelchristian) [inline bears/css/StyleLintBear.py]: it won't run unless the user provide a config file ( or that there is a .stylelintrc in the cur dir ) 
so maybe you should generate that for the user

File: bears/css/StyleLintBear.py (hunk 4 above)
Comment (Asnelchristian) [inline bears/css/StyleLintBear.py]: you can map the error sign(the cross), the warning sign...... with ``(\D)``
anyway here is the regex that worked for me
```
output_regex=r'\s*(?P<filename>.+)\s*(?P<line>\d+):(?P<column>\d+)\s*'
                  r'(\D)\s*(?P<message>.+)',
```

**Comment by Asnelchristian** [comment]
add stylelint the coala-bears/package.json file, so that it gets install before the tests

File: bears/css/StyleLintBear.py (hunk 3 above)
Comment (dracarys09) [inline bears/css/StyleLintBear.py]: I have used a setting ```stylelint_config_file``` for the same...but your point makes sense...  :smile:

**Comment by fneu** [comment]
ack https://github.com/coala/coala-bears/pull/1308/commits/d60ea9b298d4d5d3db579cc5b44e471940ef2fa2

This looks really solid to me

**Comment by fneu** [comment]
ack https://github.com/coala/coala-bears/pull/1308/commits/d60ea9b298d4d5d3db579cc5b44e471940ef2fa2

**Comment by fneu** [comment]
@rultor merge

**Pull request merged**